go-controller/pkg/node/linkmanager/link_network_manager.go

```diff
@@ -88,46 +85,36 @@ func (c *Controller) DelAddress(address netlink.Addr) error {
 	if err != nil && !util.GetNetLinkOps().IsLinkNotFoundError(err) {
 		return fmt.Errorf("no valid link associated with addresses %s: %v", address.String(), err)
 	}
+	klog.Infof("Link manager: deleting address %s from link %s", address.String(), link.Attrs().Name)
 	c.mu.Lock()
 	defer c.mu.Unlock()
+	if err := util.GetNetLinkOps().AddrDel(link, &address); err != nil {
+		if !util.GetNetLinkOps().IsLinkNotFoundError(err) {
+			return fmt.Errorf("failed to delete address %s: %v", address.String(), err)
+		}
+	}
 	c.delAddressFromStore(link.Attrs().Name, address)
-	c.reconcile()
 	return nil
 }
 
 func (c *Controller) reconcile() {
 	// 1. get all the links on the node
 	// 2. iterate over the links and get the addresses associated with it
-	// 3. cleanup any stale addresses from link that we no longer managed
-	// 4. remove any stale addresses from links that we do manage
-	// 5. add addresses that are missing from a link that we managed
+	// 3. add addresses that are missing from a link that we manage
```

Comment:
so reconcile no longer removes ips, but only adds them? Then we should change the name of the function?

Reply:
for sure.. done